Firmware code/smart_energy_meter/src/credit.hpp

```diff
@@ -1,5 +1,6 @@
-#pragma once
-
+//#pragma once
+#ifndef credit_hpp
+#define credit_hpp
```

Comment:
Why is this needed?

Reply:
The two syntaxe are doing the same things but there is a difference between them.
The #ifndef/#define/#endif trick works on any C compiler, and on some of them it speeds up the compilation process. The #pragma trick is non-standards, and only works on few C compilers, and may result in different semantics in those that do not support it.
When I use #pragma trick, I have many warning, but with  #ifndef/#define/#endif trick the warnings disappear

Reply:
but for this PR I'm just going to put back #pragma once as you wanted